scripts/installer.sh

```diff
@@ -0,0 +1,297 @@
+            $PRIVILEGED_EXECUTION "apt-get update"
+            $PRIVILEGED_EXECUTION "apt-get install -y gnupg apt-transport-https curl"
+
+            echo "# "Adding GPG public key...
+            $PRIVILEGED_EXECUTION "curl -L \"$DEBIAN_GPG_PUBLIC_KEY\" |  apt-key add -"
+
+            echo "# Adding repository to /etc/apt/sources.list"
+            case "$ODO_VERSION" in
+
+            master)
+                $PRIVILEGED_EXECUTION "echo \"deb $DEBIAN_MASTER_REPOSITORY stretch main\" |  tee -a /etc/apt/sources.list"
+                ;;
+            latest)
+                $PRIVILEGED_EXECUTION "echo \"deb $DEBIAN_LATEST_REPOSITORY stretch main\" | tee -a /etc/apt/sources.list"
+                ;;
+            *)
+                invalid_odo_version_error
+            esac
+
+            $PRIVILEGED_EXECUTION "apt-get update"
+            $PRIVILEGED_EXECUTION "apt-get install -y odo"
+        elif [ "$INSTALLER_ACTION" == "uninstall" ]; then
+            echo "# Uninstalling odo..."
+            $PRIVILEGED_EXECUTION "apt-get remove -y odo"
+        fi
+    ;;
+
+    centos|fedora)
+
+        package_manager=""
+        case "$distribution" in
+
+        fedora)
+            package_manager="dnf"
+        ;;
+        centos)
+            package_manager="yum"
+        ;;
+        esac
+
+        if [ "$INSTALLER_ACTION" == "install" ]; then
+            echo "# Installing odo version $ODO_VERSION on $distribution"
+
+            echo "# Adding odo repo under /etc/yum.repos.d/"
+            case "$ODO_VERSION" in
+
+            master)
+                $PRIVILEGED_EXECUTION "curl -L $RPM_MASTER_YUM_REPO -o /etc/yum.repos.d/bintray-odo-odo-rpm-dev.repo"
+                ;;
+            latest)
+                $PRIVILEGED_EXECUTION "curl -L $RPM_LATEST_YUM_REPO -o /etc/yum.repos.d/bintray-odo-odo-rpm-releases.repo"
+                ;;
+            *)
+                invalid_odo_version_error
+            esac
+
+            $PRIVILEGED_EXECUTION "$package_manager install -y odo"
+        elif [ "$INSTALLER_ACTION" == "uninstall" ]; then
+            echo "# Uninstalling odo..."
+            $PRIVILEGED_EXECUTION "$package_manager remove -y odo"
+        fi
+    ;;
+
+    *)
+        if [ "$INSTALLER_ACTION" == "install" ]; then
+            echo "# Could not identify distribution. Proceeding with a binary install..."
+
+            BINARY_URL=""
+            TMP_DIR=$(mktemp -d)
+            case "$ODO_VERSION" in
+
+            master)
+                BINARY_URL="$BINTRAY_URL/$platform/odo"
+                echo "# Downloading odo from $BINARY_URL"
+                curl -Lo $TMP_DIR/odo "$BINARY_URL"
+            ;;
+
+            latest)
+                BINARY_URL="$GITHUB_RELEASES_URL/odo-$platform.gz"
+                echo "# Downloading odo from $BINARY_URL"
+                curl -Lo $TMP_DIR/odo.gz "$BINARY_URL"
+                echo "# Extracting odo.gz"
+                gunzip -d $TMP_DIR/odo.gz
```

Comment:
A note that we may want to think of a way to check if `gunzip` is locally installed. Ran into issues with this.